### PR TITLE
Fix hive's drop column causes ConcurrentModificationException

### LINKED ISSUE #9733: ConcurrentModificationException when alter table drop columns
```
alter table ttd_bi_recommend_created9 drop column edndata2;
```
throw this exception

```
java.util.ConcurrentModificationException: undefined
	at java.util.ArrayList$Itr.checkForComodification(ArrayList.java:901)
	at java.util.ArrayList$Itr.next(ArrayList.java:851)
	at com.facebook.presto.hive.metastore.thrift.BridgingHiveMetastore.dropColumn(BridgingHiveMetastore.java:217)
	at com.facebook.presto.hive.metastore.CachingHiveMetastore.dropColumn(CachingHiveMetastore.java:495)
	at com.facebook.presto.hive.metastore.CachingHiveMetastore.dropColumn(CachingHiveMetastore.java:495)
	at com.facebook.presto.hive.metastore.SemiTransactionalHiveMetastore.lambda$dropColumn$9(SemiTransactionalHiveMetastore.java:357)
```

```
  @Override
    public void dropColumn(String databaseName, String tableName, String columnName)
    {
        verifyCanDropColumn(this, databaseName, tableName, columnName);
        org.apache.hadoop.hive.metastore.api.Table table = delegate.getTable(databaseName, tableName)
                .orElseThrow(() -> new TableNotFoundException(new SchemaTableName(databaseName, tableName)));
        for (FieldSchema fieldSchema : table.getSd().getCols()) {
            if (fieldSchema.getName().equals(columnName)) {
                table.getSd().getCols().remove(fieldSchema);
            }
        }
        alterTable(databaseName, tableName, table);
    }
```

I have changed the code in BridgingHiveMetastore to belows
```
    Iterator<FieldSchema> iters = table.getSd().getCols().iterator();
        while (iters.hasNext()) {
            if (iters.next().getName().equals(columnName)) {
                iters.remove();
            }
        }
```

### PR DESCRIPTION
**Pull request by ebyhr**
Fix #9733

### CODE REVIEW

File: presto-hive/src/main/java/com/facebook/presto/hive/metastore/thrift/BridgingHiveMetastore.java [hunk 1]
```diff
@@ -213,11 +213,7 @@ public void dropColumn(String databaseName, String tableName, String columnName)
         verifyCanDropColumn(this, databaseName, tableName, columnName);
         org.apache.hadoop.hive.metastore.api.Table table = delegate.getTable(databaseName, tableName)
                 .orElseThrow(() -> new TableNotFoundException(new SchemaTableName(databaseName, tableName)));
-        for (FieldSchema fieldSchema : table.getSd().getCols()) {
-            if (fieldSchema.getName().equals(columnName)) {
-                table.getSd().getCols().remove(fieldSchema);
-            }
-        }
+        table.getSd().getCols().removeIf(fieldSchema -> fieldSchema.getName().equals(columnName));
```
Comment (findepi) [inline presto-hive/src/main/java/com/facebook/presto/hive/metastore/thrift/BridgingHiveMetastore.java]: change commit message to `Fix ConcurrentModificationException when dropping Hive column`

File: presto-product-tests/src/main/java/com/facebook/presto/tests/AlterTableTests.java [hunk 2]
```diff
@@ -99,13 +99,16 @@ public void addColumn()
     @Test(groups = {ALTER_TABLE, SMOKE})
     public void dropColumn()
     {
-        query(format("CREATE TABLE %s AS SELECT n_nationkey, n_regionkey FROM nation", TABLE_NAME));
+        query(format("CREATE TABLE %s AS SELECT n_nationkey, n_regionkey, n_name FROM nation", TABLE_NAME));
```
Comment (findepi) [inline presto-product-tests/src/main/java/com/facebook/presto/tests/AlterTableTests.java]: squash tests with first commit

**Comment by ebyhr** [comment]
Thanks for your review. Updated.

**Comment by findepi** [comment]
Merged, thanks!

**Pull request merged**